pandas/io/stata.py

```diff
@@ -1494,7 +1491,11 @@ def read(self, nrows=None, convert_dates=None,
             data = data.set_index(ix)
 
         if columns is not None:
-            data = self._do_select_columns(data, columns)
+            try:
```

Comment:
actually you only have 2 uses of this (in the main body), so may not be worht it.